specification/cost-management/resource-manager/Microsoft.CostManagement/preview/2020-12-01-preview/costmanagement.generatedetailedcostreport.json

```diff
@@ -0,0 +1,405 @@
+      "get": {
+        "tags": [
+          "GenerateDetailedCostReportOperationStatus"
+        ],
+        "operationId": "GenerateDetailedCostReportOperationStatus_Get",
+        "description": "Get the status of the specified operation. This link is provided in the GenerateDetailedCostReport creation request response header.",
+        "x-ms-examples": {
+          "Get details of the operation": {
+            "$ref": "./examples/GenerateDetailedCostReportOperationStatus.json"
+          }
+        },
+        "x-ms-long-running-operation": true,
+        "parameters": [
+          {
+            "name": "operationId",
+            "in": "path",
+            "required": true,
+            "type": "string",
+            "description": "The target operation Id."
+          },
+          {
+            "$ref": "#/parameters/scopeUsageDetailsParameter"
+          },
+          {
+            "$ref": "#/parameters/apiVersionParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "OK. The request has succeeded.",
+            "schema": {
+              "$ref": "#/definitions/GenerateDetailedCostReportOperationStatuses"
+            }
+          },
+          "default": {
+            "description": "Error response describing why the operation failed.",
+            "schema": {
+              "$ref": "#/definitions/ErrorResponse"
+            }
+          }
+        }
+      }
+    }
+  },
+  "definitions": {
+    "GenerateDetailedCostReportErrorResponse": {
+      "description": "Error response indicates that the service is not able to process the incoming request. The reason is provided in the error message. \n\nSome Error responses: \n\n * 413 Request Entity Too Large - Request is throttled. The amount of data required to fulfill the request exceeds the maximum size permitted of 2Gb. Please utilize our Exports feature instead. \n\n * 429 TooManyRequests - Request is throttled. Retry after waiting for the time specified in the \"x-ms-ratelimit-microsoft.consumption-retry-after\" header. \n\n * 503 ServiceUnavailable - Service is temporarily unavailable. Retry after waiting for the time specified in the \"Retry-After\" header.",
+      "type": "object",
+      "properties": {
+        "error": {
+          "description": "The details of the error.",
+          "$ref": "#/definitions/ErrorDetails"
+        }
+      }
+    },
+    "GenerateDetailedCostReportDefinition": {
+      "description": "The definition of a cost detailed report.",
+      "properties": {
+        "metric": {
+          "description": "The type of the detailed report. By default ActualCost is provided",
+          "type": "string",
+          "enum": [
+            "ActualCost",
+            "AmortizedCost"
+          ],
+          "x-ms-enum": {
+            "name": "GenerateDetailedCostReportMetricType",
+            "modelAsString": true
+          }
+        },
+        "timePeriod": {
```

Comment:
What happens when a POST operation has timePeriod and billingPeriod? which one wins? consider adding this info as part of the description.

Reply:
If the customer has both timePeriod and billingPeriod, API will return back a 400 and ask the client to only use one. This is what I have in description for them, let me know if I should update it "Can only have one of either timePeriod or invoiceId or billingPeriod parameters."